In [1]:
from sklearn.datasets import load_iris
iris=load_iris()
x=iris.data
y=iris.target


In [3]:
from sklearn.preprocessing import StandardScaler
scaler=StandardScaler()
x_scaled=scaler.fit_transform(x)

In [5]:
from sklearn.cluster import KMeans

kmeans = KMeans(n_clusters=3, random_state=42, n_init=10)
cluster_labels = kmeans.fit_predict(x_scaled)


### Silhouette Score

In [6]:
from sklearn.metrics import silhouette_score

sil_score = silhouette_score(x_scaled, cluster_labels)
print(sil_score)


0.45994823920518635


### Inertia

In [7]:
inertia = kmeans.inertia_
print(inertia)


139.82049635974982


### Adjusted Rand Index (using true labels)

In [9]:
from sklearn.metrics import adjusted_rand_score

ari = adjusted_rand_score(y, cluster_labels)
print(ari)


0.6201351808870379


# K-Means Clustering Analysis: Iris Dataset

This report summarizes the performance of the K-Means clustering algorithm (k=3) applied to the scaled Iris dataset.

### 1. Model Evaluation Metrics
| Metric | Result |
| :--- | :--- |
| **Silhouette Score** | **0.460** |
| **Inertia** | **139.82** |
| **Adjusted Rand Index (ARI)** | **0.620** |


### 2. Result Interpretation
* **Cluster Separation:** The **Silhouette Score of 0.46** suggests that the model has achieved decent separation between the three clusters. While not perfectly isolated, the groups are distinct enough to be useful.
* **Cluster Compactness:** The **Inertia** value indicates that the data points are relatively close to their respective centroids, suggesting consistent groupings.
* **Validation against Ground Truth:** The **ARI of 0.62** is the most significant result. It confirms a strong similarity between the unsupervised clusters generated by the model and the actual true labels (species), proving the model effectively captured the natural structure of the data.

### 3. Final Conclusion
The K-Means algorithm successfully identified the three species groups with high accuracy. The combination of a solid ARI and a decent Silhouette score indicates that feature scaling was effective and that the choice of $k=3$ was appropriate for this dataset.
